ERCS/erc-7945.md

```diff
@@ -0,0 +1,250 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-05-09
+requires: 20
+---
+
+## Abstract
+
+This proposal draws up a standard interface of confidential transaction supported token contracts, by providing basic functionality without loss of generality. Contracts following the standard can provide confidentiality for users’ balances and token transfer value.
+
+## Motivation
+
+A standard interface allows confidential transactions of tokens on Ethereum (and/or other EVM-compatible blockchains) to be applied by certain parties which are sensitive to transfer amount, or by privacy-preserving applications.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+### Contract Interface
+
+#### Methods
+
+##### name
+
+Returns the name of the token - e.g. `"MyConfidentialToken"` .
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+```Solidity
+function name() public view returns (string)
+```
+
+##### symbol
+
+Returns the symbol of the token. e.g. `"cHIX"` .
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+```Solidity
+function symbol() public view returns (string)
+```
+
+##### decimals
+
+Returns the number of decimals the token uses - e.g. `8` , means to divide the token amount by `100000000` to get its user representation.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+```Solidity
+function decimals() public view returns (uint8)
+```
+
+##### confidentialBalanceOf
+
+Returns the account confidential balance of another account with address `owner` .
+
+```Solidity
+function confidentialBalanceOf(address owner) 
+public view returns (bytes memory confidentialBalance)
+```
+
+##### confidentialTransfer
+
+Transfers `value` amount of tokens (behind `_confidentialTransferValue` ) to address `_to` , and MUST fire the `ConfidentialTransfer` event. The function SHOULD `throw` if the message caller’s `_proof` of this transfer fails to be verified.
+
+Note:
+
+* Callers MUST handle `false` from returns (bool success). Callers MUST NOT assume that `false` is never returned!
+* Implementations can fully customize the proof system, (de)serialization strategies of `bytes` and/or the business workflow. For example, when implementing "Zether" (DOI: 10.1007/978-3-030-51280-4_23) confidential token contracts, the `_confidentialTransferValue` and accounts' confidential balances will be encrypted homomorphically under ElGamal public keys, and `_proof` will consist 3 parts to check:
```

Comment:
Please use the CSL-JSON format [defined in EIP-1](https://eips.ethereum.org/EIPS/eip-1#digital-object-identifier-system) for citations using a DOI.

Reply:
> Please use the CSL-JSON format [defined in EIP-1](https://eips.ethereum.org/EIPS/eip-1#digital-object-identifier-system) for citations using a DOI.

DOI format issue fixed